frontend/public/components/storage-class-form.tsx

```diff
@@ -0,0 +1,927 @@
+        },
+        storageMode: {
+          name: 'Storage Mode',
+          values: {thinProvisioned: 'ThinProvisioned', thickProvisioned: 'ThickProvisioned'},
+          placeholder: 'Storage Provision Mode'
+        },
+        secretRef: {
+          name: 'Secret Reference',
+          required: true,
+          placeholder: 'Reference to a configured Secret object'
+        },
+        readOnly: {
+          name: 'Read Only',
+          type: 'checkbox'
+        },
+        fsType: {
+          name: 'Filesystem Type',
+          placeholder: 'Filesystem to use for the volume'
+        }
+      }
+    },
+    storageOs: {
+      title: 'StorageOS',
+      provisioner: 'kubernetes.io/storageos',
+      documentationLink: 'https://kubernetes.io/docs/concepts/storage/storage-classes/#storageos',
+      parameters: {
+        pool: {
+          name: 'Pool',
+          placeholder: 'Name of the StorageOS distributed capacity pool from which to provision the volume'
+        },
+        description: {
+          name: 'Description',
+          placeholder: 'Description to assign to volumes that were created dynamically'
+        },
+        fsType: {
+          name: 'Filesystem Type',
+          placeholder: 'Default filesystem type to request'
+        },
+        adminSecretName: {
+          name: 'Admin Secret Name',
+          placeholder: 'Name of the secret to use for obtaining the StorageOS API credentials'
+        },
+        adminSecretNamespace: {
+          name: 'Admin Secret Namespace',
+          placeholder: 'Namespace where the API configuration secret is located',
+          required: (params) => {
+            const adminSecretName = _.get(params, 'adminSecretName.value', null);
+            return adminSecretName !== null && adminSecretName !== '';
+          },
+        },
+      }
+    }
+  };
+
+  reclaimPolicies = {
+    Retain: 'Retain',
+    Delete: 'Delete'
+  };
+
+  componentDidUpdate(prevProps) {
+    if (this.props !== prevProps) {
+      this.props.watchK8sList(this.reduxId, {}, StorageClassModel);
+      const loaded = this.props.k8s.getIn([StorageClassModel.plural, 'loaded']);
+      if (loaded) {
+        const data = this.props.k8s.getIn([StorageClassModel.plural, 'data']);
+        this.resources = {
+          data: data && data.toArray().map(p => p.toJSON()),
+          loadError: this.props.k8s.getIn([StorageClassModel.plural, 'loadError']),
+          loaded: loaded,
+        };
+
+        this.validateForm();
+      }
+    }
+  }
+
+  setParameterHandler = (param, event, checkbox) => {
+    let newParams = {...this.state.newStorageClass.parameters};
+    if (checkbox) {
+      newParams[param] = {value: event.target.checked};
+    } else {
+      if (event.target) {
+        newParams[param] = {value: event.target.value};
+      } else {
+        newParams[param] = {value: event};
+      }
+    }
+
+    for (let key in newParams) {
+      if (newParams.hasOwnProperty(key)) {
```

Comment:
Done.

Reply:
I'm not sure `hasOwnProperty` check is not necessary with `_.each`... I didn't think so, but it's not clear from the docs best I can tell.

You might use `_.forOwn` instead to avoid needing the check

Reply:
Good call. Done.